pkiinit/pkiInit.go

```diff
@@ -18,51 +18,75 @@ package main
 import (
 	"flag"
 	"fmt"
+	"log"
 	"os"
+
+	"github.com/edgexfoundry/security-secret-store/pkiinit/option"
 )
 
 type exit interface {
 	callExit(int)
 }
 
-type exitCode struct {
-	exit
+type exitCode struct{}
+
+type optionDispatcher interface {
+	run() (int, error)
 }
 
-var exitInstance = newExit(&exitCode{})
+type pkiInitOptionDispatcher struct{}
 
-func main() {
-	var helpOpt bool
-	// only help option for this phase
-	if flag.Lookup("h") == nil {
-		// only register once if it is empty
-		// to prevent "redefined flag error"
-		flag.BoolVar(&helpOpt, "h", false, "help message")
-		flag.BoolVar(&helpOpt, "help", false, "help message")
-	}
+var exitInstance = newExit()
+var dispatcherInstance = newOptionDispatcher()
+var helpOpt bool
+var generateOpt bool
 
+func init() {
```

Comment:
where is this init() called? can't find it

Reply:
init() is built-in hook func for Golang to call during the package initialilzation. In this case main package level.  It will get called first before main() func is invoked and it is called automatically by Golang executable.